src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs

```diff
@@ -0,0 +1,93 @@
+﻿using CommunityToolkit.Maui.Core;
+
+namespace CommunityToolkit.Maui.Alerts;
+
+/// <inheritdoc/>
+public partial class Toast : IToast
+{
+	/// <inheritdoc/>
+	public string Text { get; set; } = string.Empty;
+
+	/// <inheritdoc/>
+	public ToastDuration Duration { get; set; }
+
+	/// <summary>
+	/// Create new Toast
+	/// </summary>
+	/// <param name="message">Toast message</param>
+	/// <param name="duration">Toast duration</param>
+	/// <returns>New instance of Toast</returns>
+	public static IToast Make(
+		string message,
+		ToastDuration duration = ToastDuration.Short)
+	{
+		return new Toast()
+		{
+			Text = message,
+			Duration = duration
+		};
+	}
+
+	/// <summary>
+	/// Show Toast
+	/// </summary>
+	public virtual partial Task Show(CancellationToken token = default);
+
+	/// <summary>
+	/// Dismiss Toast
+	/// </summary>
+	public virtual partial Task Dismiss(CancellationToken token = default);
+
+#if !(IOS || ANDROID || MACCATALYST || WINDOWS)
+	/// <summary>
+	/// Show Toast
+	/// </summary>
+	public virtual partial Task Show(CancellationToken token)
+	{
+		token.ThrowIfCancellationRequested();
+		return Task.CompletedTask;
+	}
+
+	/// <summary>
+	/// Dismiss Toast
+	/// </summary>
+	public virtual partial Task Dismiss(CancellationToken token)
+	{
+		token.ThrowIfCancellationRequested();
+		return Task.CompletedTask;
+	}
+#endif
+
+	/// <summary>
+	/// Dispose Toast
+	/// </summary>
+	public async ValueTask DisposeAsync()
+	{
+		await DisposeAsyncCore();
+		GC.SuppressFinalize(this);
+	}
+
+	/// <summary>
+	/// Dispose Toast
+	/// </summary>
+	protected virtual async ValueTask DisposeAsyncCore()
+	{
+#if ANDROID
+		await Device.InvokeOnMainThreadAsync(() => nativeToast?.Dispose());
+#else
+		await Task.CompletedTask;
+#endif
+	}
+
+#if IOS || MACCATALYST || WINDOWS
+	static TimeSpan GetDuration(ToastDuration duration)
+	{
+		return duration switch
+		{
+			ToastDuration.Short => TimeSpan.FromSeconds(2),
+			ToastDuration.Long => TimeSpan.FromSeconds(3.5),
+			_ => throw new ArgumentOutOfRangeException(nameof(duration), duration, null)
```

Comment:
Let's throw an `InvalidEnumArgumentException`:

```suggestion
			_ => throw new InvalidEnumArgumentException(nameof(Duration), (int)duration, typeof(ToastDuration))
```